plugins/scaffolder-actions/scaffolder-backend-module-utils/src/actions/jsonata/yaml.ts

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2022 Larder Software Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { createTemplateAction } from '@backstage/plugin-scaffolder-backend';
+import jsonata from 'jsonata';
+import { resolveSafeChildPath } from '@backstage/backend-common';
+import fs from 'fs-extra';
+import yaml from 'js-yaml';
+
+export function createYamlJSONataTransformAction() {
+  return createTemplateAction<{
+    path: string;
+    expression: string;
+    writeOutputPath?: string;
+    options?: {
+      indent: number;
+    };
+  }>({
+    id: 'roadiehq:utils:jsonata:yaml:transform',
+    description:
+      'Allows performing jsonata opterations and transformations on a YAML file in the workspace. The result can either be read from the `result` step output or be written back to a file using the `writeOutputPath` property.',
+    schema: {
+      input: {
+        type: 'object',
+        required: ['path', 'expression'],
+        properties: {
+          path: {
+            title: 'Path',
+            description: 'Input path to read yaml file.',
+            type: 'string',
+          },
+          expression: {
+            title: 'Expression',
+            description: 'JSONata expression to perform on the input',
+            type: 'string',
+          },
+          writeOutputPath: {
+            title: 'Output Path',
+            description: 'If set this will write the serialized yaml here.',
+            type: 'string',
+          },
```

Comment:
We provide that option also. There is a `roadiehq:utils:fs:write` action that can be used on the output of the yaml serialize action.

This action allows you to open, edit, and write the file contents in a single action step. Its more concise I suppose.

Reply:
Sorry, I just don't really see why this action got a special treatment where it can do multiple things and not others?

Reply:
I think I might be misunderstanding the question.

This action basically reads a yaml file, edits the contents and allows the data to be written back to a file.

What do you mean when you say "do multiple things"?

Reply:
> This action basically reads a yaml file, edits the contents and allows the data to be written back to a file.

Why have this got the option to write it back? Why don't we let the user decide if they want to write it back, print it or do whatever they want with the output.

The optional boolean caught my eye. If it is true, this action can write to the fs, if it is false it doesn't write to the fs and it always reads, edits. So it does two different things based on the variable. For me it would be cleaner if this just returns the edited value into the output variable and then in a next step we can write it out if we want to

Reply:
ok. cool. will do.